src/library/scala/collection/mutable/LinkedHashMap.scala

```diff
@@ -129,6 +132,59 @@ class LinkedHashMap[K, V]
     }
   }
 
+  override def filterInPlace(p: (K, V) => Boolean): this.type = {
+    if (size > 0) {
+      var bucket = 0 // the index of the current bucket in the hash table
+      val tableTable = table.table
+      val buckets = tableTable.length
+      var newFirstEntry: Entry = null // will become the new firstEntry
+      var earlier: Entry = null // the entry "before" the current entry, may cross bucket boundaries
+      var newTableSize = table.tableSize
+
+      while (bucket < buckets) {
+        (tableTable(bucket) : @switch) match {
```

Comment:
I don't understand why you're switching on a match statement instead of using a simple conditional for the non-null case.